lib/core/string.cpp

```diff
@@ -101,6 +101,14 @@ std::string to_string(const std::chrono::duration<Rep, Period> &duration) {
   using Clock = std::chrono::system_clock;
 
   std::ostringstream oss;
+
+#ifdef _WIN32
+  // Windows' time functions (e.g. gmtime) don't support datetimes before 1970.
+  if (duration < std::chrono::duration<Rep, Period>::zero()) {
+    return "(cannot format datetime)";
```

Comment:
Maybe?
```suggestion
    return "(datetime before epoch, cannot format)";
```

Reply:
I'm slightly cautious about making this too long - maybe over cautious? Not sure on this...

Reply:
Datetimes tend to be too long to display properly in space-limited cases anyway (like HTML repr, plots). So this would make things worse.
Also, telling users that something can't be done without giving any explanation would just frustrate them / make them mad at us.